.github/workflows/browserstack-run.yaml

```diff
@@ -0,0 +1,126 @@
+        required: true
+      BROWSERSTACK_ACCESS_KEY:
+        description: BrowserStack Access Key
+        required: true
+      SLACK_WEBHOOK:
+        description: Slack Notifier Incoming Webhook
+        required: true
+
+jobs:
+  browserstack-run:
+    runs-on: ubuntu-latest
+    outputs:
+      browserstack_build_id: ${{ steps.start_run.outputs.browserstack_build_id }}
+
+    steps:
+      # Get the test artifacts prepared in previous step
+      - name: Get the app-debug-signed.apk artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: app-debug-signed.apk
+
+      - name: Get the forgerock-integration-tests-debug-androidTest.apk BrowserStack artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: forgerock-integration-tests-debug-androidTest-signed.apk
+
+      - name: Unzip app-debug-signed.apk and forgerock-integration-tests-debug-androidTest-signed.apk
+        run: |
+          unzip -o app-debug-signed.apk || echo "Skipping unzip for app-debug"
+          unzip -o forgerock-integration-tests-debug-androidTest-signed.apk || echo "Skipping unzip for test apk"
+
+      # Upload app to BrowserStack
+      - name: Upload app-debug-signed.apk to BrowserStack
+        id: upload_app
+        run: |
+          APP_UPLOAD_RESPONSE=$(curl -u "${{ secrets.BROWSERSTACK_USERNAME }}:${{ secrets.BROWSERSTACK_ACCESS_KEY }}" \
+            -X POST "https://api-cloud.browserstack.com/app-automate/espresso/v2/app" \
+            -F "file=@app-debug-signed.apk")
+          echo "APP_URL=$(echo $APP_UPLOAD_RESPONSE | jq -r '.app_url')" >> $GITHUB_ENV
+
+      
+      # Upload test suite APK to BrowserStack
+      - name: Upload forgerock-integration-tests-debug-androidTest-signed.apk to BrowserStack
+        id: upload_test_suite
+        run: |
+          TEST_UPLOAD_RESPONSE=$(curl -u "${{ secrets.BROWSERSTACK_USERNAME }}:${{ secrets.BROWSERSTACK_ACCESS_KEY }}" \
+            -X POST "https://api-cloud.browserstack.com/app-automate/espresso/v2/test-suite" \
+            -F "file=@forgerock-integration-tests-debug-androidTest-signed.apk")
+          echo "TEST_SUITE_URL=$(echo $TEST_UPLOAD_RESPONSE | jq -r '.test_suite_url')" >> $GITHUB_ENV
+
+
+      # Prepare BrowserStack run configuration. 
+      # To run tests in parallel on multiple devices, use the "shards" option. Example:
+      #      "shards": {
+      #        "numberOfShards": 3,
+      #        "deviceSelection": "all"
+      #      }
+      #
+      - name: Prepare BrowserStack run configuration
+        run: |
+          DEVICES_JSON=$(echo '${{ vars.BS_DEVICES }}' | jq -R 'split(",")')
+          cat <<EOF > browserstack-run-config.json
+          {
+            "app": "${{ env.APP_URL }}",
+            "testSuite": "${{ env.TEST_SUITE_URL }}",
+            "singleRunnerInvocation": ${{ vars.BS_SINGLE_RUNNER_INVOCATION }},
+            "devices": $DEVICES_JSON,
+            "deviceLogs": ${{ vars.BS_DEVICE_LOGS }},
```

Comment:
`BS_DEVICE_LOGS`, `BS_NETWORK_LOGS`, `BS_VIDEO` are repo variables - all set to `false` by default.
For more details see the [BrowserStack API reference docs](https://www.browserstack.com/docs/app-automate/api-reference/espresso/builds)...